src/codegen/codegen_neuron_cpp_visitor.cpp

```diff
@@ -147,6 +152,41 @@ void CodegenNeuronCppVisitor::print_point_process_function_definitions() {
 }
 
 
+void CodegenNeuronCppVisitor::print_check_table_function_prototypes() {
+    if (info.table_count == 0) {
+        return;
+    }
+
+    // print declarations of `check_*` functions
+    for (const auto& function: info.functions_with_table) {
+        auto name = function->get_node_name();
+        auto internal_params = internal_method_parameters();
+        printer->fmt_line("void {}{}({});",
+                          table_function_prefix(),
+                          method_name(name),
+                          get_parameter_str(internal_params));
+    }
+
+    // definition of `_check_table_thread` function
+    // signature must be same as the `nrn_thread_table_check_t` type
+    printer->fmt_line(
+        "static void {}(Memb_list* _ml, size_t id, Datum* _ppvar, Datum* _thread, "
+        "NrnThread* _nt, int _type, _nrn_model_sorted_token const& _sorted_token)",
```

Comment:
There are utilities in place to help with this:
* ParamVector
* get_parameter_str
and I feel the list is sufficiently long to use them. I'm aware there's other places where we don't either.